index.js

```diff
@@ -120,235 +112,250 @@ module.exports = function generate(options) {
     return result;
   }
 
+  var cwd = process.cwd();
   var assertionsPattern = options.assertions || [
     'assertions/*/*.md',
     'assertions.md'
   ];
   var output = options.output || 'site-build';
-
-  metalSmith('.')
-    .destination(output)
-    .source('documentation')
-    .use(
-      require('metalsmith-collections')({
-        assertions: {
-          pattern: assertionsPattern
-        },
-        apiPages: {
-          pattern: ['api/*.md', 'api.md']
-        },
-        pages: {
-          pattern: ['*.md', '!assertions.md', '!api.md']
-        }
-      })
-    )
-    .use(
-      require('./lib/include-static-assets')({
-        path: path.resolve(__dirname, 'static')
-      })
-    )
-    // Dynamicly generate metadata for assertion files
-    .use(function(files, metalsmith, next) {
-      Object.keys(files)
-        .filter(function(file) {
-          return /\.md$/.test(file);
+  var tmpOutput = `${output}-tmp`;
```

Comment:
Could this be put below `require('os').tmpdir()` to avoid leaving turds in CWD if the script gets aborted or fails? (If we can move it there, the dir should probably also contain some randomness or `process.pid`).

Reply:
Good shout - have done this in https://github.com/unexpectedjs/unexpected-documentation-site-generator/pull/43/commits/5d83f1965c9f0a1c593f691313e8e32cc47629bc.